### ISSUE TITLE
Clarify (un)signed-ness of pointers for comparisons

### ISSUE THREAD

**Issue by gamozolabs** [rust-lang/rust#77497]
It seems currently pointer comparisons, ala. `(foo as *const u32) > (bar as *const u32)`. Emits an unsigned comparison. I looked a little bit to see if this was documented somewhere and I didn't see anything super obvious.

Is this something that is defined behavior, and if it is, could we clarify it in the primitive `pointer` type documentation? This can be relevant especially in many kernel designs when there are "negative" addresses in the kernel and "positive" in user-space.

Thanks!
-B

**Comment by RalfJung** [comment]
Yes, pointers are definitely treated as unsigned integers for inequality comparisons. There is no such thing as a "negative address", just an address >= 0xF0000000[00000000].
